### ISSUE TITLE
Questions on MCMC Baselines

### ISSUE THREAD

**Issue by jeevster** [plainerman/Variational-Doob#2]
Cool work! Some questions regarding the MCMC baselines:
1. Can you share the references on which the 1-way and 2-way MCMC shooting baselines are based?
2. How was the initial path for alanine dipeptide constructed (seems like you are loading a pdb trajectory but not sure where it comes from)?

Thanks!

**Comment by plainerman** [comment]
Thank you for raising this issue.

1) The code we have used is basically a re-implementation of the relevant parts of http://openpathsampling.org/latest/ see [this](https://pubs.acs.org/doi/10.1021/acs.jctc.8b00626) and [this](https://pubs.acs.org/doi/10.1021/acs.jctc.8b00627) paper.

This is a good [youtube video](https://www.youtube.com/watch?v=ys26j1YnERk) about the topic.

2) To get the initial pdb trajectory, we simply ran MD simulation for alanine dipeptide with a high temperature (I think it is 500K) and then took the first trajectory that (roughly) connects the two states.

I hope this clarifies your questions!

**Comment by jeevster** [comment]
Thanks!

**Comment by plainerman** [comment]
Just let me know if there is anything else I can help with :)

**Comment by jeevster** [comment]
Thanks! Another general question regarding the Doobs method. When testing the method on a new real-world molecular system, how would you generally suggest we choose the following parameters: `num_gaussians`, `trainable_weights`, 'internal_coordinates', and 'resnet'. Can you give a quick rundown on whether certain combinations of these settings lead to more expressive and/or expensive training, and what the tradeoffs associated with these (and any other important hyperparameters) are? I understand that this is very vague/open-ended question but just trying to get a general feel, thanks!

**Comment by plainerman** [comment]
`num_gaussians`: For each Gaussian you basically find one transition channel (compare Figure 3 from the paper where we show 1 vs. 2 Gaussians)
`trainable_weights`: I would set it to true for basically all real-world systems. It learns different weights for the transition channels. It doesn't really introduce much overhead. However, I am not sure if we implemented this for all cases. If not, it should throw an error but should be easy enough to implement.  (compare $w$ in Equation 17 of the paper). 
`internal_coordinates`: Represents the molecule with bond angles etc., which have a nice distribution and are easier to learn. We only implemented this for alanine dipeptide. You should use it if you have a good representation of your molecule. But we got it working with Chignolin without.
`resnet`: Introduces skip connections to the neural network, making it easier to train deep neural networks. I guess you can always use it. 

`num_gaussians` impacts the training time a lot. We never ran experiments with more than 10, maybe for smaller systems you could do more.

We are working on a more efficient and stable training setup right now.

**Comment by jeevster** [comment]
Thanks, this is very helpful.
One more question, regarding the Number of Potential Energy Evaluations metric reported in the code/paper. Since this metric is based on the batch size/number of epochs of training, how do you determine when to stop training/when a plausible transition path has been achieved?

**Comment by plainerman** [comment]
For an unknown system it can be tricky I guess. In principle, any of the hyperparameters (architecture, layers, batch size, learning rate, num_gaussians...) could make it harder to train or let the algorithm converge to different paths.

Chemists probably have a better intuition about sensible energy levels, transition barriers, and maybe even about the behavior of the system. For somebody without that knowledge - or when you have an unknown system - your only way is to look if the loss has converged and then look at the results.
